### ISSUE TITLE
Speed up Mac OS CI

### ISSUE THREAD

**Issue by AdamGleave** [HumanCompatibleAI/imitation#536]
The Mac OS CI executor is currently taking 4x longer than our Linux executor. This seems to be for two main reasons: 
  1. The `brew install` step alone takes 9m, to install just a handful of packages. We might be able to cache this to speed things up. I also question whether some of the packages are necessary.
  2. It's a smaller executor than the others (medium and not x-large). Unfortunately CircleCI only goes up to large for MacOS and that's not available to us without upgrading to an annual plan. I think the best option is to run multiple MacOS executors in parallel and shard the tests across them using e.g. [pytest-shard](https://pypi.org/project/pytest-shard/)

**Comment by AdamGleave** [comment]
@Rocamonde @taufeeque9 one of you should probably take this on? Juan you seemed excited/annoyed by Mac OS so feel free to take this if it scratches an itch, but otherwise probably better for @taufeeque9 to look into this as you'll be busy with the EPIC project soon.

**Comment by Rocamonde** [comment]
@AdamGleave I agree that this is an issue, right now macOS checks are absurdly slow. I spent a bit of time looking into this issue as I use macOS for all my work and found this to be quite surprising. Unfortunately, I did not manage to make much progress. I'll look into it more and contribute it asynchronously if I come up with anything, but I think that it might be a long fix and given my other project commitments it might work out best if someone else takes the lead, unless no active contributor uses macOS, as that would make it a bit hard for them to test stuff locally (I'd be surprised, but it would also be interesting to know).
